### PR TITLE
FLOW-9147

### PR DESCRIPTION
**Pull request by aarostegui**
New endpoint to export CSV

## Summary

This is an endpoint to export a CSV file with the matchmaking results

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings (ESLint)

## Links

-   Jira: https://flowfact.atlassian.net/browse/FLOW-9147

Thanks so much for your PR, your contribution is appreciated! ❤️

### CODE REVIEW

File: src/service/MatchmakingService/MatchController.ts [hunk 1]
```diff
@@ -75,6 +75,24 @@ export default class MatchController extends APIClient {
...
     }
 
+    /**
+     * Fetch matches for a given search profile.
+     * @param searchProfile
+     *      A search profile entity
+     * @param query
+     *      The query to filter special results
+     */
+     async matchForSearchProfileWithBodyAndReportCSV(
```
Comment (ffjanhoeck) [inline src/service/MatchmakingService/MatchController.ts:85]: When it fetches something, then please prefix it with `fetch`.
Isn't `fetchMatchings` enough ? This name sounds really complicated

File: src/service/MatchmakingService/MatchController.ts [hunk 2]
```diff
@@ -75,6 +75,24 @@ export default class MatchController extends APIClient {
...
+        searchProfile: Entity,
+        query: MatchmakingTypes.FilterQuery = {},
+    ) {
+        return await this.invokeApiWithErrorHandling(`/match/search-profile/csv`, 'POST', searchProfile, {
```
Comment (ffjanhoeck) [inline src/service/MatchmakingService/MatchController.ts:89]: Please define return type! 

```suggestion
        return await this.invokeApiWithErrorHandling<???>(`/match/search-profile/csv`, 'POST', searchProfile, {
```

**Review by ffjanhoeck — CHANGES_REQUESTED** [review]

File: src/service/MatchmakingService/MatchController.ts (hunk 1 above)
Comment (aarostegui) [reply src/service/MatchmakingService/MatchController.ts:85]: Maybe fetchMatchingAsCSV?

File: src/service/MatchmakingService/MatchController.ts (hunk 2 above)
Comment (aarostegui) [reply src/service/MatchmakingService/MatchController.ts:89]: Not sure what to do here, the return type is a CSV file, maybe a string?

Comment (ffjanhoeck) [reply src/service/MatchmakingService/MatchController.ts]: Is it really just the CSV content as string or do you download a file here ?

File: src/service/MatchmakingService/MatchController.ts (hunk 1 above)
Comment (ffjanhoeck) [reply src/service/MatchmakingService/MatchController.ts]: `AsCSV` should be the type. I mean, the type should define that

File: src/service/MatchmakingService/MatchController.ts (hunk 2 above)
Comment (aarostegui) [reply src/service/MatchmakingService/MatchController.ts]: We download the file contents with this endpoint, it's content type is text/csv

**Review by ffjanhoeck — APPROVED** [review]

**Pull request merged**